tests/failpoints/cases/test_disk_full.rs

```diff
@@ -0,0 +1,209 @@
+}
+
+fn fail_leader_full(cluster: &mut Cluster<ServerCluster>) {
+    fail::cfg(DISK_FULL_LEADER, "return").unwrap();
+
+    let leader = cluster.leader_of_region(1).unwrap();
+
+    {
+        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
+        let index_1 = raft_stat.get_last_index();
+
+        let key = String::from("8000").into_bytes();
+        let value = String::from("8000").into_bytes();
+        let rx = cluster.async_put(&key, &value).unwrap();
+        is_error_response(&rx.recv_timeout(Duration::from_secs(10)).unwrap());
+
+        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
+        let index_2 = raft_stat.get_last_index();
+        assert!(index_1 == index_2);
+    }
+
+    {
+        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
+        let index_1 = raft_stat.get_last_index();
+
+        let split_key = String::from("0000").into_bytes();
+        let region = cluster.get_region(&split_key);
+        let split_count_before = cluster.pd_client.get_split_count();
+        cluster.split_region(&region, &split_key, Callback::None);
+        let split_count_after = cluster.pd_client.get_split_count();
+        // println!(
+        //     "fail split count: {} {}",
+        //     split_count_before, split_count_after
+        // );
+        assert!(split_count_before == split_count_after);
+
+        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
+        let index_3 = raft_stat.get_last_index();
+        assert!(index_1 == index_3);
+    }
+
+    {
+        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
+        let index_1 = raft_stat.get_last_index();
+
+        let left_key = String::from("5555").into_bytes();
+        let right_key = String::from("7777").into_bytes();
+        let region1 = cluster.get_region(&left_key);
+        let region2 = cluster.get_region(&right_key);
+        assert!(region1.get_id() != region2.get_id());
+        let resp = cluster.try_merge(region1.get_id(), region2.get_id());
+        assert!(is_error_response(&resp));
+
+        let raft_stat = cluster.raft_local_state(1, leader.get_store_id());
+        let index_4 = raft_stat.get_last_index();
+        assert!(index_1 == index_4);
+    }
+
+    fail::remove(DISK_FULL_LEADER);
+}
+
+fn success_leader_full(cluster: &mut Cluster<ServerCluster>) {
+    let key = String::from("5555").into_bytes();
+    fail::cfg(DISK_FULL_LEADER, "return").unwrap();
+    cluster.must_get(&key).unwrap();
+
+    let region = cluster.get_region(&key);
+    let old_leader = cluster.leader_of_region(region.get_id()).unwrap();
+    let peers = region.get_peers();
+
+    let target = peers
+        .into_iter()
+        .find(|x| x.get_store_id() != old_leader.get_store_id())
+        .unwrap();
+
+    cluster.must_transfer_leader(region.get_id(), (*target).clone());
+    let new_leader = cluster.leader_of_region(region.get_id()).unwrap();
```

Comment:
`must_transfer_leader` already contains the compare logic.

Reply:
done